### PR TITLE
Add pycbc support for the snowline sampler

### PR DESCRIPTION
**Pull request by ahnitz**
The snowline sampler (https://github.com/JohannesBuchner/snowline) is for unimodal distributions but may be good for some cases where we have very high SNR signals. (or want a quick initial estimate).

### CODE REVIEW

**Comment by cdcapano** [comment]
@ahnitz Let me know when you want me to review this.

**Comment by ahnitz** [comment]
@cdcapano Basically now, but I'm waiting for the CI to check if there are any issues introduced at that level.

**Comment by ahnitz** [comment]
@cdcapano Can we merge this? I think there is one cc small thing I can fix, but otherwise, does this seem Ok to go?

**Review by cdcapano — APPROVED** [review]
A few requests, mostly related to documentation. Approving, but please address my questions before merging.

File: pycbc/inference/sampler/snowline.py
Comment (ahnitz) [reply pycbc/inference/sampler/snowline.py:119]: Noted. It's not implemented at the moment. I'm not sure right now how to hook into the underlying code, but I would guess it might be possible in a similar way to dynesty (e.g. pickle) at the very least.

File: pycbc/pool.py
Comment (ahnitz) [reply pycbc/pool.py:128]: Ok, added. It's so that we can use imap for our generic pool objects. That allows writing scripts that can be parallelized with imap and also not fail for the single core case.

**Pull request merged**
